src/AppBundle/Security/ApiKeyUserProvider.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace AppBundle\Security;
+
+use Symfony\Component\Security\Core\User\UserProviderInterface;
+use Symfony\Component\Security\Core\User\User;
+use Symfony\Component\Security\Core\User\UserInterface;
+use Symfony\Component\Security\Core\Exception\UnsupportedUserException;
+use Doctrine\Common\Persistence\ManagerRegistry;
+
+class ApiKeyUserProvider implements UserProviderInterface
+{
+    /**
+     * @var ManagerRegistry
+     */
+    private $registry;
+
+    /**
+     * @param ManagerRegistry $registry
+     */
+    public function __construct(ManagerRegistry $registry)
+    {
+        $this->registry = $registry;
+    }
+
+    /**
+     * @param string $apiKey
+     * @return null|string
+     */
+    public function getUsernameByApiKey($apiKey)
```

Comment:
rename to ```getUserByApiKey``` fix doc block and return object. We programming in OOP paradigm, are you remember?